lib/security/openssl_misc_utils.pm

```diff
@@ -55,7 +55,8 @@ sub install_openssl {
             install_11_workaround_when_no_legacy();
             return;
         }
-        add_suseconnect_product('sle-module-legacy');
+        record_info('Extensions List', script_output("SUSEConnect --list-extensions"));
+        add_suseconnect_product('sle-module-legacy') unless is_sled;
```

Comment:
This was broken also before, but this is redundant of `has_no_legacy_module`
IMO there should be `else` to empty line 53